documentation/CAMARA-API-access-and-user-consent.md

```diff
@@ -379,6 +379,14 @@ More details about the standard flow can be found in the official IETF specifica
 
 The purpose of this document section is to standardise the specification of `securitySchemes` and `security` across all CAMARA API subprojects with common mandatory guidelines as agreed by the Technical Steering Committee (TSC) and the participants of this Working Group.
 
+CAMARA guidelines define a set of authorization flows which can grant API Consumers access to the API.
+Which specific authorization flows are to be used will be determined during the onboarding process, happening between the API Consumer (the direct API invoker) and the API producer exposing the API. When API access for an API consumer is ordered,  the declared purpose for accessing the API can be taken into account. This is also being subject to the prevailing legal framework dictated by local legislation and eventually also considers the capabilities of the application (frontend and backend) ultimately involved in the API invocation flow. 
+The possible authorization flows should be configured on the API specification, eventually being purpose dependent. 
```

Comment:
```suggestion 
```
I suggest to remove this sentence.

>The possible authorization flows should be configured on the API specification

@Elisabeth-Ericsson What do you mean by that? I think that sentence could be misleading, since it could be understood that the authorization flows should be included in the API specification YAML file, which is the opposite of what is documented in this section. This topic was a long and closed discussion. And this section of the document is intended to document the agreement reached on what securitySchemes and security refer to for the CAMARA API specification.

>eventually being purpose dependent.

IMHO, as I said in https://github.com/camaraproject/IdentityAndConsentManagement/issues/108#issuecomment-1895620024, I don't think the authorization flow has a direct relationship with the purpose. The allowed purpose(s) set during the onboarding process will actually determine the legal basis applied, which depends on the local legislation. And ultimately, it may determine whether an opt-in (consent capture)/opt-out is required.

Reply:
1) I know that the agreement is to not have the possible authorization flows in the YAML file. So where is then the information coming from ? Obviously the API producer must indicate, which flows are supported per API. Would be the  term "API documentation offered by the API producer" be more favorable ?
2) The purpose dependency is introduced in the original text of the document. I have not added this, but just kept it.
I'm happy to remove, but this requires that the author of the text agrees.

Reply:
> 1. I know that the agreement is to not have the possible authorization flows in the YAML file. So where is then the information coming from ? Obviously the API producer must indicate, which flows are supported per API. Would be the  term "API documentation offered by the API producer" be more favorable ?

I would just delete the sentence. As @eric-murray said, CAMARA should just focus on providing the technical mechanisms that allow API Providers to offer API products that are compliant with local regulation. The text already says that _"specific authorization flows are to be used will be determined during the onboarding process, happening between the API Consumer (the direct API invoker) and the API producer exposing the API"_. We do not need to say more in CAMARA.

> 2. The purpose dependency is introduced in the original text of the document. I have not added this, but just kept it.
>    I'm happy to remove, but this requires that the author of the text agrees.

[Existing document content](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md#camara-api-specification---authorization-and-authentication-common-guidelines) does not include that  and as far as I can see in the commit history it was added in b9d93193e5f5e3b660c571f34384bb7c23242fe4.

Reply:
It is correct that I have added the paragraph in https://github.com/camaraproject/IdentityAndConsentManagement/commit/b9d93193e5f5e3b660c571f34384bb7c23242fe4, but the statement related to the purpose is a copy from below.
Please check chapter **Mandatory template for info.description in CAMARA API specs**. when you scroll to the right (almost not readable) then you see the original statement. Mine is just a copy.

So we need to fix then both places then. Is this ok ?

Reply:
I actually agree with the documented statement:

>Authorization and authentication_
>
>CAMARA guidelines defines a set of authorization flows which can grant API clients access to the API functionality, as outlined in the document [CAMARA-API-access-and-user-consent.md](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md). Which specific authorization flows are to be used will be determined during onboarding process, happening between the API Client and the Telco Operator exposing the API, taking into account the declared purpose for accessing the API, while also being subject to the prevailing legal framework dictated by local legislation.
>
>It is important to remark that in cases where personal user data is processed by the API, and users can exercise their rights through mechanisms such as opt-in and/or opt-out, the use of 3-legged access tokens becomes mandatory. This measure ensures that the API remains in strict compliance with user privacy preferences and regulatory obligations, upholding the principles of transparency and user-centric data control.

Sorry Eli, maybe I missed it or misunderstood you, but where is it written _"The possible authorization flows should be configured on the API specification, eventually being purpose dependent. "_?

Reply:
Hi Jesus, 
the text above does not contain a statement about where possible authorization flows should be configured. This was my addition, which got removed by now.
However the statement related to the purpose refers to the sentence: "... _taking into account the declared purpose for accessing the API_".
This sentence is extracted from the info.description template, which is available at the end of the document. This statement means that the selected authflow can be purpose dependent.
Do you have a different interpretation of this sentence ?

Reply:
I think that in the end, both interpretations are valid. In essence, it does not really matter.  It gets easier, if we do not have to consider purpose dependency. So perfectly fine with me.
The critical sentence is gone anyway...
Hope that the pull request can be approved now.